packages/shared/src/lib/utils/invoiceTotals.ts

```diff
@@ -27,9 +28,22 @@ export const calculateInvoiceTotals = (data: InvoiceItem[]) => {
   };
 };
 
-export const calculateItemTotal = (item: InvoiceItem): number => {
-  const discountAmount = item.discount;
-  const priceAfterDiscount = item.unitPrice - discountAmount;
+export const calculateItemTotal = (
+  item: InvoiceItem,
+  options: { format?: boolean; currencyDecimal?: number } = {}
+): number => {
+  const { format = false, currencyDecimal = 2 } = options;
+
+  const discountAmount = format
+    ? // @ts-expect-error
+      parseFloat(formatUnits(item.discount, currencyDecimal))
```

Comment:
We must do all arithmetic before formatting into human-readable format.

parseFloat after formatUnits introduces risk of quantization errors (because `float` loses precision at very large values.